static/app/views/replays/detail/focusArea.tsx

```diff
@@ -29,11 +30,21 @@ function FocusArea(props: Props) {
 function ActiveTab({active, event, eventWithSpans}: Props & {active: TabBarId}) {
   const {routes, router} = useRouteContext();
   const organization = useOrganization();
-
+  // memoize the memory events coming from the event because we reassign
+  // the data without the memory events within the 'performance' case
+  const memorySpans = useMemo(
+    () => eventWithSpans?.entries[0]?.data?.filter(datum => datum?.data?.memory) || [],
+    [eventWithSpans?.entries[0]]
+  );
   switch (active) {
     case 'console':
       return <div id="console">TODO: Add a console view</div>;
     case 'performance':
+      if (eventWithSpans) {
+        eventWithSpans.entries[0].data = eventWithSpans?.entries[0]?.data?.filter(
+          datum => !datum?.data?.memory
+        );
+      }
```

Comment:
Would love some feedback on this bit here. I couldn't figure out a nice way to filter out memory events so I memoized them prior to removing them from the data being passed to EventEntry. It feels gross so I opted to get eyes on it.

I'm not opposed to filtering within the EventEntry component but I wasn't sure if modifying another component for the sake of an experimental feature was kosher. Though to me that feels like the cleaner way to handle this.

Reply:
Hmm I would probably handle this in `useReplayEvent` where we create the merged span in the first place. Thoughts on this @ryan953 ?

Reply:
I agree that `useReplayEvent` seems like the place to put this, shouldn't need useMemo or anything in this case. 

It is data that's derived from the state, and the hook will only update when state changes. So you can filter everything right at the `return` statement and it shouldn't run more often than it needs to.

We probably should have the hook return data that's pre-filtered for each use case, so each component gets exactly what it needs. Maybe the exported names would endup something like this:
```
return {
  fetchError: state.fetchError,
  fetching: state.fetching,
  onRetry,
  event,

  // ideally:
  chapterBreadcrumbs,
  performanceSpans,
  memorySpans,
};
```

Reply:
I dig it, I'll make it happen.

Reply:
Made a type for the memory spans called MemorySpanType and adjusted code to reflect that throughout the implementation. Hook now surfaces the memorySpans